packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -637,6 +637,43 @@ export const dropSearchIndex = (
   };
 };
 
+export const runAggregateSearchIndex = (
+  name: string
+): IndexesThunkAction<void> => {
+  return function (_dispatch, getState, { globalAppRegistry }) {
+    const {
+      searchIndexes: { indexes },
+      namespace,
+    } = getState();
+    const searchIndex = indexes.find((x) => x.name === name);
+    if (!searchIndex) {
+      return;
+    }
+    const pipeline = JSON.stringify([
+      {
+        $search: {
+          index: name,
+          text: {
+            query: 'string',
+            path: 'string',
+          },
+        },
+      },
+    ]);
+    globalAppRegistry?.emit('search-indexes-run-aggregate', {
+      ns: namespace,
+      aggregation: {
+        pipelineText: pipeline,
+        autoPreview: true,
+        name: null,
+        id: null,
+        collationString: null,
+        comments: null,
+      },
```

Comment:
Depending on how much time we have for this, it might make sense to allow to pass pipeline with the metadata (we already support `pipeline` and `pipelineText` being passed to the `configureStore` in aggregation plugin, just don't support actually propagating them through events all the way to configure store call) in addition to stored pipeline so that we don't need to fake the stored pipeline format here, this feels a bit fragile

Reply:
Yeah that makes sense. I'll change this and pass it as a pipelineText. I was not sure of that tbh, but it does make sense.